source/Messages.cpp

```diff
@@ -88,5 +100,36 @@ void Messages::Reset()
 {
 	lock_guard<mutex> lock(incomingMutex);
 	incoming.clear();
-	list.clear();
+	recent.clear();
+	log.clear();
+}
+
+
+
+// Get color that should be used for drawing messages of given importance.
+const Color *Messages::GetColor(Importance importance)
+{
+	const Color *color;
```

Comment:
This variable is unnecessary, just `return` immediately in the switch.

Reply:
If we were to remove this variable, we would need to move the null check for getting the default color out of this method and put it everywhere this method is used.

Reply:
Can't you put it after the switch as an unconditional return?

Reply:
I mean, the null check is unnecessary anyways because every enum value is covered

Reply:
Isn't it needed for when the needed color is not defined?

Reply:
Not if you use Get as quyykk suggests below.

Reply:
Okay, but when a message color isn't defined, we want to use the default message color, which isn't necessarily white.

Reply:
But if one of the switch case colors isn't defined, with what certainty can we say that the default color is defined?
I'd prefer to use Get and have the color display incorrectly if it isn't defined than to attempt to fall back on some default that we're similarly unsure of whether it exists. (This is a similar case as with #9140 where we shouldn't be trying to hide content data errors.)

Reply:
Here it's not hiding data errors, it's how it works currently: some colors are defined, and some aren't, because they are supposed to use the default color. But of course it can be changed, especially considering that AFAIK nothing else works like this.

Reply:
The use of a default colour when drawing the messages in Engine seems particularly strange given that the provided default colour is identical to the colour that is produced by the default constructor of Color, that is, it is no different to if Get were being used and no Color with that name was ever loaded.